iot_config/src/client/mod.rs

```diff
@@ -35,6 +35,34 @@ pub struct Client {
     batch_size: u32,
 }
 
+macro_rules! call_with_retry {
+    ($rpc:expr) => {{
+        use tonic::Code;
+
+        let mut attempt = 1;
+        loop {
+            let response = $rpc.await;
+            match response {
+                Ok(resp) => break Ok(resp),
+                Err(status) => match status.code() {
+                    Code::Cancelled | Code::DeadlineExceeded | Code::Unavailable => {
+                        if attempt < 3 {
+                            attempt += 1;
+                            tokio::time::sleep(Duration::from_millis(attempt * 100)).await;
```

Comment:
delay between attempts is very short.  Doesnt give much time for any transitory issue to resolve

Reply:
@andymck What do you think it should be?

Reply:
5 secs ?

Reply:
We default the rpc timeout in the client endpoint builder to 5 sec already so it should probably be less then that since we’re making three calls but I can skip the increasing back off  and just retry 3 times at the default. That reminds me though we’d talked about modifying the request to set the timeout header which is going to shoot this whole thing down

Reply:
Should we start with 1 second, so then do 1,2,3??

Reply:
works for me